avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailModuleWizardStep.kt

```diff
@@ -31,24 +31,50 @@
  */
 package org.availlang.intellij.plugin.module
 
+import avail.anvil.environment.GlobalEnvironmentSettingsV1
 import com.intellij.ide.util.projectWizard.ModuleWizardStep
+import com.intellij.openapi.module.Module
 import javax.swing.JComponent
-import javax.swing.JLabel
 
 /**
  * The [ModuleWizardStep] used to set up a new Avail project.
  *
  * @author Richard Arriaga
  */
-class AvailModuleWizardStep: ModuleWizardStep()
+class AvailModuleWizardStep(builder: AvailModuleBuilder) : ModuleWizardStep()
 {
-	override fun getComponent(): JComponent
+
+	init
 	{
-		return JLabel("TODO!")
+		// this listener is called after you choose a location and a name
+		// even though we ask for the Avail module roots and config first
+	  builder.addListener { module ->
+			AvailProjectTemplate.create(module.config)
+		}
 	}
 
+	private val Module.config get() =
+		createProjectPanel.let { panel ->
+			AvailProjectTemplate.Config(
+				projectLocation = moduleFilePath.substringBeforeLast("/"),
+				fileName = name,
+				rootsDir = panel.rootsDirField.input.ifEmpty { "roots" },
+				rootName = panel.rootNameField.input.ifEmpty { name },
+				importStyles = panel.importStyles.isSelected,
+				importTemplates = panel.importTemplates.isSelected,
```

Comment:
The common practice in this kind of plugin seems to be having the initial project layout copied from the `resources` folder into the `target` module folder. When they do this, they support Velocity templates, which help variable replacements in case any parts of the project layout are dynamic. The velocity templates are then evaluated, and the final layout is created.